python/simpler/worker.py

```
@@ -1865,6 +1865,120 @@ class _StartupCancelled(BaseException):
+                if error is not None:
+                    raise error
+                return
+            self._wait_in_progress = True
+            run_id = self._run_id
+
+        assert run_id is not None
+        native_error: BaseException | None = None
+        try:
+            remaining = None if deadline is None else max(0.0, deadline - time.monotonic())
+            completed = self._worker._wait_run_handle(run_id, remaining)
+        except Exception as exc:  # native run failures are terminal
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)